consensus/src/epoch_manager.rs

```diff
@@ -132,6 +133,7 @@ pub struct EpochManager<P: OnChainConfigProvider> {
     commit_state_computer: Arc<dyn StateComputer>,
     storage: Arc<dyn PersistentLivenessStorage>,
     safety_rules_manager: SafetyRulesManager,
+    sys_txn_providers: Vec<Arc<dyn SysTxnProvider>>,
```

Comment:
you probably don't need shared ownership here?

Reply:
I'm not sure. But something might be easier if it's shared ownership.
E.g., at the beginning of every epoch, we need to send a clone of this vec from epoch manager to the newly created proposal manager.

Reply:
I think you can either send a &, or just let proposal manager owns it?

Reply:
Arc is what we've been doing unless we can figure out how to convince compiler the lifetime of round manager is shorter than epoch manager